src/BSON/Javascript.c

```diff
@@ -46,6 +46,49 @@ PHONGO_API zend_class_entry *php_phongo_javascript_ce;
 
 zend_object_handlers php_phongo_handler_javascript;
 
+/* Initialize the object from a string and return whether it was successful. */
+static bool php_phongo_javascript_initialize(php_phongo_javascript_t *intern, const char *javascript, phongo_zpp_char_len javascript_len, zval *scope TSRMLS_DC)
+{
+	if (scope && Z_TYPE_P(scope) != IS_OBJECT && Z_TYPE_P(scope) != IS_ARRAY && Z_TYPE_P(scope) != IS_NULL) {
+		return false;
+	}
+
+	intern->javascript = estrndup(javascript, javascript_len);
+	intern->javascript_len = javascript_len;
```

Comment:
If you mean just the length bit? I disagree - as it's handy when strnduping it to return to the user. And the value is "free".